### DISCUSSION TITLE
Issue using CDSM in SOLWEIG

### DISCUSSION THREAD

**Discussion by bweeding** [UMEP-dev/UMEP#274] [Q&A]
I'm having issues with my CDSM in SOLWEIG. The vegetation doesn't appear properly in the model, and I'm guessing this might be linked to the sky view factors generated. I have included screenshots of the CDSM and sky view factor, and included a zip of the .las and met file used. The driving code can be found [here](https://github.com/bweeding/phd/blob/main/las_fusion_processing_draft3_uni.py).

For the CDSM to work properly with SOLWEIG, should its values be metres above sea level, or metres above the DEM? Or is there another issue here?

Cheers
Ben

![image](https://user-images.githubusercontent.com/57697604/113817173-40fd0600-97b9-11eb-97b0-01399edf4fa1.png)

![image](https://user-images.githubusercontent.com/57697604/113817196-4b1f0480-97b9-11eb-9655-eef3de864c91.png)

[solweig_cdsm_issues.zip](https://github.com/UMEP-dev/UMEP/files/6269291/solweig_cdsm_issues.zip)

**Comment by biglimp** [comment]
By the look of it you have an issue that all non-vegetation pixels are not zero. It seems like building pixels are (probably) nodata which will end up as e.g. -9999. IT also seems like you also have included ground elevation which should not in there. A CDSM consist of either vegetation oixels (meter above ground) or non vegetation pixels (zero).

**Reply by bweeding** [reply]
Ah thank you, I will investigate the generation of the CDSM and attempt to find a solution.

**Comment by biglimp** [comment]
Maybe you can find some hints from this tutorial. https://umep-docs.readthedocs.io/projects/tutorial/en/latest/Tutorials/LidarProcessing.html#lidarprocessing

**Reply by bweeding** [reply]
Ah yep I think I have followed that procedure, but there may be a problem with the way my LIDAR data has been classified, I'm checking with the author.

**Reply by bweeding** [reply]
Just double checking here, following the Lidar processing tutorial it instructs you to construct the CDSM from both vegetation points and ground points:
![image](https://user-images.githubusercontent.com/57697604/114108984-9f3ffb00-9917-11eb-819b-7d1fd01a611e.png)
However you've said above that the CDSM should either contain vegetation points, or zeros. Is this second piece of advice the correct idea?

**Reply by biglimp** [reply]
Yes, then everything is normalized so that everything will become meters above ground with the optional parameter where you add your ground.dtm.

**Reply by bweeding** [reply]
Ok great! 

When I'm creating the DEM/DSM/CDSM from lidar data (in .las format), do you recommend going from .dtm to .tif, or .asc to .tif?

**Reply by biglimp** [reply]
I use dtm to tif. Never tested asc to tif

**Comment by bweeding** [comment]
Ah ok, using fusion as in the tutorial?

**Reply by biglimp** [reply]
yes
